react/src/components/molecules/LinkDropdown/index.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import is from 'is';
+import classNames from 'classnames';
+import ButtonWithIcon from '../../atoms/buttons/ButtonWithIcon';
+import DecorativeLink from '../../atoms/links/DecorativeLink';
+import Icon from '../../atoms/icons/Icon';
+import './style.css';
+
+class LinkDropdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      buttonExpand: false
+    };
+    this.wrapperRef = React.createRef();
+    this.setDropDownButtonRef = this.setDropDownButtonRef.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+  }
+  componentDidMount() {
+    this.buttonClicked = false;
+    document.addEventListener('mousedown', this.handleClickOutside);
+    this.dropDownButtonRef.addEventListener('mousedown', this.handleRefMouseDown);
+  }
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+    this.dropDownButtonRef.removeEventListener('mousedown', this.handleRefMouseDown);
+  }
+  setDropDownButtonRef(node) {
+    this.dropDownButtonRef = node;
+  }
+  closeDropdown() {
+    this.setState({ buttonExpand: false });
+  }
+  handleClick() {
+    this.setState((prevState) => ({ buttonExpand: !prevState.buttonExpand }));
+  }
+  handleSelect(event) {
+    const { target } = event;
+    this.setState({ buttonExpand: false });
+    if (is.fn(this.props.onItemSelect)) {
+      this.props.onItemSelect({ target });
+    }
+  }
+  handleClickOutside(event) {
+    // Close the panel if the user clicks outside the component.
+    const node = this.wrapperRef.current;
+    if ((node && !node.contains(event.target))) {
+      if (this.state.buttonExpand) {
+        this.setState({ buttonExpand: false });
+      }
+    }
+  }
+  render() {
+    const { dropdownItems, dropdownButton } = this.props;
+    const dropdownButtonProps = {
+      ...dropdownButton,
```

Comment:
should `'aria-haspopup': ` in this component also be true, without the option to overwrite?

Reply:
always set to true